src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyRelayedResponse.java

```diff
@@ -126,43 +126,52 @@ private void onResponseBegin(
+                if ((statusGroup == 2 || statusGroup == 3) && prefer != null)
                 {
-                    hs.item(h -> h.name(ACCESS_CONTROL_EXPOSE_HEADERS).value(ETAG));
+                    hs.item(h -> h.name(PREFERENCE_APPLIED).value(prefer));
+                    hs.item(h -> h.name(ACCESS_CONTROL_EXPOSE_HEADERS).value(PREFERENCE_APPLIED));
+
+                    if (hasEtag)
+                    {
+                        hs.item(h -> h.name(ACCESS_CONTROL_EXPOSE_HEADERS).value(ETAG));
+                    }
                 }
-            }
-        };
+            };
 
-        Flyweight.Builder.Visitor mutator = (b, o, l) ->
-            factory.httpBeginExRW.wrap(b, o, l)
-                         .typeId(httpBeginEx.typeId())
-                         .headers(headers)
-                         .build()
-                         .sizeof();
+            Flyweight.Builder.Visitor mutator = (b, o, l) ->
+                factory.httpBeginExRW.wrap(b, o, l)
+                                     .typeId(httpBeginEx.typeId())
+                                     .headers(headers)
+                                     .build()
+                                     .sizeof();
 
-        final MutableDirectBuffer writeBuffer = factory.writeBuffer;
-        final BeginFW newBegin = factory.beginRW.wrap(writeBuffer, 0, writeBuffer.capacity())
-                .routeId(receiverRouteId)
-                .streamId(receiverReplyId)
-                .traceId(traceId)
-                .authorization(authorization)
-                .affinity(affinity)
-                .extension(e -> e.set(mutator))
-                .build();
+            final MutableDirectBuffer writeBuffer = factory.writeBuffer;
+            final BeginFW newBegin = factory.beginRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                                                    .routeId(receiverRouteId)
+                                                    .streamId(receiverReplyId)
+                                                    .traceId(traceId)
+                                                    .authorization(authorization)
+                                                    .affinity(affinity)
+                                                    .extension(e -> e.set(mutator))
+                                                    .build();
 
-        factory.router.setThrottle(receiverReplyId, this::onResponseMessage);
-        receiver.accept(newBegin.typeId(), newBegin.buffer(), newBegin.offset(), newBegin.sizeof());
+            factory.router.setThrottle(receiverReplyId, this::onResponseMessage);
+            receiver.accept(newBegin.typeId(), newBegin.buffer(), newBegin.offset(), newBegin.sizeof());
+        }
+        else
+        {
+            factory.writer.send500(receiver, receiverRouteId, receiverReplyId, traceId);
+            factory.router.clearThrottle(receiverReplyId);
+            factory.counters.responses.getAsLong();
+        }
```

Comment:
What happens if the response headers actually included `500` status?
This logic assumes the specific default and hard codes the response, which is not quite what is intended.